### PR TITLE
Enable container tests in Windows CI

### LINKED ISSUE #55102: Enable container support for Windows CI workflows
Context:
While we currently have active Windows validation in our CI pipeline, it lacks container capabilities. As a result, any integration tests that depend on containers cannot be run on Windows in CI.

Goal:
Enable container execution support within the Windows CI pipeline so that container-dependent integration tests can run seamlessly across all primary target operating systems.

Benefits:

- Closing the Container Testing Gap on Windows: Ensures that our container-based integration tests are validated on Windows instead of being skipped.
- Catching OS-Specific Container Interactions: Catches edge cases specifically related to how container runtimes interact with Windows hosts.

Definition of Done

- Container execution support is successfully enabled and stable within the Windows CI environment.
- Container-dependent integration tests execute successfully within the Windows CI workflows without infrastructure-related failures.

### PR DESCRIPTION
**Pull request by arnabnandy7**
This enables container-dependent integration tests in the Windows CI workflow. The Windows integration-test lane now uses the existing container test arguments, verifies that the runner supports Linux containers before starting the build, and cleans up Docker resources afterward.

The workflow was validated with `actionlint`. The test matrix JSON was parsed successfully, and `git diff --check` completed without errors.

* Fixes #55102

### CODE REVIEW

**Comment by arnabnandy7** [comment]
@geoand can you please review?

**Comment by geoand** [comment]
Thanks for this! TBH,  our CI is pretty unstable now, and I would prefer we don't introduce more complexity until we stabilize it.

Hopefully that will happen soon

**Comment by arnabnandy7** [comment]
> Thanks for this! TBH,  our CI is pretty unstable now, and I would prefer we don't introduce more complexity until we stabilize it.
> 
> Hopefully that will happen soon 

@geoand Should I mark it draft till then?

**Comment by geoand** [comment]
Yeah

**Comment by gsmet** [comment]
Were you able to get Docker to work on Windows GitHub runners?

Because I tried to get it working there: https://github.com/quarkusio/quarkus/pull/55177/changes and it wasn't able to find the Docker environment.

FWIW, it used to not be working and was documented as such. Thus why it was disabled and from what I can see, Testcontainers fail to find a working Docker installation.

File: .github/matrix-jvm-tests.json [hunk 1]
```diff
@@ -104,10 +104,11 @@
         "tag": "jvm-tests-integration",
         "java-version": 17,
         "java-distribution": "temurin",
-        "maven_args": "-DskipDocs -Dformat.skip",
+        "maven_args": "$JVM_TEST_MAVEN_ARGS",
         "maven_opts": "-Xmx2g -XX:MaxMetaspaceSize=1g",
         "os-name": "windows-latest",
-        "runs-on": false,
+        "runs-on": true,
+        "container-support": true,
```
Comment (gsmet) [inline .github/matrix-jvm-tests.json:111]: Ah I see what you did there. It's not going to work in forks though. I'll try to see if I can get something to work.

Comment (arnabnandy7) [reply .github/matrix-jvm-tests.json:111]: Thanks 🙏

**Comment by arnabnandy7** [comment]
@gsmet try a small WSL2-based experiment using the WSL distro IP for both `DOCKER_HOST` and `TESTCONTAINERS_HOST_OVERRIDE`, rather than Windows localhost. 

First run a minimal Java Testcontainers smoke test, then enable only integration-test modules that do not require host bind mounts. If that remains unreliable, move the coverage to a dedicated periodic Windows runner rather than the regular PR matrix.
